Fixed orders panel flickering

octaedro opened the pull request:
Fixes #5609

This PR fixes the orders panel flickering when Gutenberg is activated.

### Screenshots
![Screen Capture on 2020-11-17 at 14-14-38](https://user-images.githubusercontent.com/1314156/99423458-55f6fc80-28df-11eb-8f1d-f738f43a947c.gif)

### Detailed test instructions:

- Be sure the `Gutenberg` plugin is installed and activated.
- Go to the `Home` screen.
- Verify no flickering happens in the order panel.
- Create a few orders (manually or using the [Smooth generator](https://github.com/woocommerce/wc-smooth-generator)).
- Go to `Analytics` > `Settings` and select different combinations of the checkboxes under `Actionable Statuses` (**don't select unregistered statuses**).
- Go back to the `Home` screen.
- Again, verify there isn't any flickering in the order panel.

<!--- Note: When displaying information based on sample data, such as SwaggerHub, 
be sure to detail parts affected in Release Notes --->

### Changelog Note:
Fix: Fixed orders panel flickering
<!--- Optional: Enter a changelog note following the WooCommerce core format using prefixes of Enhancement:, Tweak:, Dev:, Fix:, Performance:. If no note is entered, one will be constructed from the title and labels. --->

octaedro committed:
Fixed orders panel flickering

This commit fixes the orders panel flickering when Gutenberg is activated

octaedro committed:
Fixed badge flickering

joshuatf commented on client/homescreen/activity-panel/orders/index.js:
```diff
@@ -361,10 +361,17 @@ export default withSelect( ( select, props ) => {
+	if (
+		countUnreadOrders === null ||
+		typeof reportOrders === 'undefined' ||
+		typeof actionableOrders === 'undefined' ||
```
Can `actionableOrders` be `undefined`?  Looks like an array is created above.

octaedro replied:
Nice catch @joshuatf! We don't need that check. It was removed in the commit dcac5b1

joshuatf commented on client/homescreen/activity-panel/orders/index.js:
```diff
@@ -304,8 +304,8 @@ export default withSelect( ( select, props ) => {
 
-	if ( ! countUnreadOrders ) {
-		return { isRequesting: countUnreadOrders !== 0 };
+	if ( ! orderStatuses.length && countUnreadOrders === 0 ) {
```
Curious why we separate this logic from the condition below.  It's a little hard to follow, but it seems both are related to whether or not we're still requesting.

octaedro replied:
If we know that at least one `order status` was selected and there aren't orders to show, it's not necessary to continue with the other requests that we do later in the code, [here](https://github.com/woocommerce/woocommerce-admin/blob/b0600d7071ea0f333cff7dc969b8e57297e7b572/client/homescreen/activity-panel/orders/index.js#L319) and [here](https://github.com/woocommerce/woocommerce-admin/blob/b0600d7071ea0f333cff7dc969b8e57297e7b572/client/homescreen/activity-panel/orders/index.js#L355).
So in order to have a more efficient behavior, I added this check before doing those calls to avoid continuing when we already have all the data that we need.
But for sure the logic is a little confusing and a future refactor would be a good idea.

joshuatf replied:
>But for sure the logic is a little confusing and a future refactor would be a good idea.

➕ 1️⃣

joshuatf reviewed (commented):
Very nice!  This PR fixes the flickering in the orders panel.  Left a couple questions around the organization of logic, but otherwise testing well.

octaedro committed:
Removed unnecessary line

octaedro committed:
Fixed items selector

This commit fixes the items selector. The method 'getItemsTotalCount' was returning the 'defaultValue' instead of the 'totalCount' when  its value was 0

octaedro commented:
Thank you @joshuatf for your review. I added the change you suggested and another small fix.
Could you give this PR another view 👀 ?

joshuatf reviewed (approved):
Testing well and seems to resolve any flickering.  Thanks for the fix here, Fernando!  LGTM 🚢

octaedro commented:
Cherry picked into `release/1.7.2`
